lib/private/Mail/Mailer.php

```diff
@@ -304,6 +305,17 @@ protected function getSmtpInstance(): EsmtpTransport {
 
 			$currentStreamingOptions = array_merge_recursive($currentStreamingOptions, $streamingOptions);
 
+			/** @psalm-suppress InternalMethod */
+			$stream->setStreamOptions($currentStreamingOptions);
+		} else if ($mailSmtphost === 'localhost' || $mailSmtphost === '127.0.0.1') {
+			$currentStreamingOptions = $stream->getStreamOptions();
+			$currentStreamingOptions = array_merge_recursive($currentStreamingOptions, array(
+				'ssl' => array(
+					'allow_self_signed' => true,
+					'verify_peer' => false,
+					'verify_peer_name' => false
+				)
+			));
```

Comment:
I'm of two minds here - one is that it indeed makes localhosts work without any setup, which, as you stated has confused many users due to how the Symfony Mailer works. OTOH I'm not entirely happy with overwriting config options per default.

I think there's some middle ground here: check if the `ssl` option is set and only overwrite it if a) the host is `localhost || 127.0.0.1` and b) if the option doesn't already exist in the `$currentStreamingOptions`

I also think it doesn't need to be an elseif but can be an if condition on it's own. Something like:

```suggestion
		} 
		$local = ['localhost', '127.0.0.1']; // You could make this a constant
		if (in_array($mailSmtphost, $local) && !isset($currentStreamingOptions['ssl'])) {
			$currentStreamingOptions = $stream->getStreamOptions();
			$currentStreamingOptions = array_merge_recursive($currentStreamingOptions, array(
				'ssl' => array(
					'allow_self_signed' => true,
					'verify_peer' => false,
					'verify_peer_name' => false
				)
			));
```